src/Dapr.Actors/Reentrancy/ActorLogicalReentrancyContext.cs

```diff
@@ -0,0 +1,36 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Actors.Reentrancy
+{
+    using System;
+    using System.Threading;
+
+    internal static class ActorLogicalReentrancyContext
+    {
+        private static readonly AsyncLocal<string> fabActAsyncLocal = new AsyncLocal<string>();
+
+        public static bool IsPresent()
```

Comment:
The other change you should make here is to a use a class (not a string) with a property that holds the string.

This allows you to *wipe out* the reentrancy value when the invocation ends. 

Here's an example of this pattern: https://github.com/aspnet/HttpAbstractions/blob/master/src/Microsoft.AspNetCore.Http/HttpContextAccessor.cs#L12

The problem with the current approach is that it's possible (and easy) to start background tasks from inside an actor method that capture the current reentrancy state.

Ex:

```C#
public async Task MyActorThing() 
{
      _ = Task.Run(async () =>
     {
             await Task.Delay(TimeSpan.FromMinutes(5));
             
             // if you use an ActorProxy here, it would be fooled into thinking this is a  re-entrant call, even though it's been 5 minutes...
      });
}
```